pkg/fluxApplication/FluxApplicationService.go

```diff
@@ -0,0 +1,148 @@
+			}
+			req.Clusters = append(req.Clusters, clusterConfig)
+		}
+	}
+	applicationStream, err := impl.helmAppClient.ListFluxApplication(ctx, req)
+
+	return applicationStream, err
+}
+
+func (impl *FluxApplicationServiceImpl) ListFluxApplications(ctx context.Context, clusterIds []int, w http.ResponseWriter) {
+	appStream, err := impl.listApplications(ctx, clusterIds)
+	impl.pump.StartStreamWithTransformer(w, func() (proto.Message, error) {
+		return appStream.Recv()
+	}, err,
+		func(message interface{}) interface{} {
+			return impl.appListRespProtoTransformer(message.(*gRPC.FluxApplicationList))
+		})
+}
+
+func (impl *FluxApplicationServiceImpl) appListRespProtoTransformer(deployedApps *gRPC.FluxApplicationList) bean.FluxAppList {
+	appList := bean.FluxAppList{ClusterId: &deployedApps.ClusterId}
+
+	fluxApps := make([]bean.FluxApplication, 0)
+
+	for _, deployedapp := range deployedApps.FluxApplication {
+
+		fluxApp := bean.FluxApplication{
+			Name:                  deployedapp.Name,
+			HealthStatus:          deployedapp.HealthStatus,
+			SyncStatus:            deployedapp.SyncStatus,
+			ClusterId:             int(deployedapp.EnvironmentDetail.ClusterId),
+			ClusterName:           deployedapp.EnvironmentDetail.ClusterName,
+			Namespace:             deployedapp.EnvironmentDetail.Namespace,
+			FluxAppDeploymentType: deployedapp.FluxAppDeploymentType,
+		}
+		fluxApps = append(fluxApps, fluxApp)
+	}
+	appList.FluxApps = &fluxApps
+	return appList
+}
+func (impl *FluxApplicationServiceImpl) DecodeFluxAppId(appId string) (*bean.FluxAppIdentifier, error) {
+	return DecodeFluxExternalAppAppId(appId)
+}
+func (impl *FluxApplicationServiceImpl) GetFluxAppDetail(ctx context.Context, app *bean.FluxAppIdentifier) (*bean.FluxApplicationDetailDto, error) {
+	config, err := impl.helmAppService.GetClusterConf(app.ClusterId)
+	if err != nil {
+		impl.logger.Errorw("error in getting cluster config", "appIdentifier", app, "err", err)
+		return nil, fmt.Errorf("failed to get cluster config for app %s in namespace %s: %w", app.Name, app.Namespace, err)
+	}
+	fluxDetailResponse, err := impl.getFluxAppDetailTree(ctx, config, app)
+	if err != nil {
+		impl.logger.Errorw("error in getting Flux app detail tree", "appIdentifier", app, "err", err)
+		return nil, fmt.Errorf("failed to get Flux app detail tree for app %s in namespace %s: %w", app.Name, app.Namespace, err)
+	}
+
+	appDetail := &bean.FluxApplicationDetailDto{
+		FluxApplication: &bean.FluxApplication{
+			Name:                  app.Name,
+			SyncStatus:            fluxDetailResponse.FluxApplication.SyncStatus,
+			HealthStatus:          fluxDetailResponse.FluxApplication.HealthStatus,
+			Namespace:             app.Namespace,
+			ClusterId:             app.ClusterId,
+			FluxAppDeploymentType: fluxDetailResponse.FluxApplication.FluxAppDeploymentType,
+			ClusterName:           fluxDetailResponse.FluxApplication.EnvironmentDetail.GetClusterName(),
+		},
```

Comment:
Are we completely sure that in fluxDetailResponse , FluxApplication will not be nil in any case, or EnvironmentDetail inside FluxApplication obj. Please make sure that nil is handled if a nil is expected.